keepassxc-browser/background/client.js

```diff
@@ -82,8 +83,12 @@ keepassClient.sendNativeMessage = function(request, enableTimeout = false, timeo
         const listener = ((port, action) => {
             const handler = (msg) => {
                 if (msg && msg.action === action) {
+                    // If the request has a separate requestID, check if it matches when there's no nonce (an error message)
+                    const isNotificationOrError = !msg.nonce
+                        && (!request.requestID && !msg.requestID || (request.requestID && msg.requestID && request.requestID === msg.requestID))
```

Comment:
This is mind bending, recommend splitting this apart. I actually think it is enough to just check for no nonce and equality between the requestID's. If they are both null then it still returns true (the first set of && conditions) and if not null they return true when equal (the second set of conditions).

Reply:
Fixed the both brain fogs.